mimic/test/behavior_tests.py

```diff
@@ -0,0 +1,213 @@
+"""
+Automatically generate tests for behavior registration/deletion APIs.
+Also contains helper functions for specific behavior testing.
+"""
+import json
+
+from uuid import UUID, uuid4
+
+from six import text_type, string_types
+
+from twisted.trial.unittest import SynchronousTestCase
+
+from zope.interface import Attribute, Interface
+
+from mimic.test.helpers import json_request, request_with_content
+
+
+class IBehaviorAPITestHelper(Interface):
+    """
+    Helper class that provides some setup and assertion methods that tests
+    for behavior CRUD need.  An instance will be created and
+    used for each CRUD test method.
+    """
+    root = Attribute("The root resource for mimic.")
+    behavior_api_endpoint = Attribute(
+        "The API endpoint for behaviors for the event to be tested.")
+    criteria = Attribute("""
+    A list of criteria for the behaviors to be injected.  Example::
+
+        [
+            {"criteria1": "regex_pattern.*"},
+            {"criteria2": "regex_pattern.*"},
+        ]
+    """)
+    name_and_params = Attribute("""
+    An list of 1 or 2 tuples of name and parameters, which together
+    with the criteria, can form a behavior specification.  Any more than
+    2 will ignored.  Example::
+
+        # tuples of (name of the behavior, parameters for the behavior)
+        [
+            ("fail", {"code": 500, "message": "Stuff is broken"}),
+            ("fail", {"code": 404, "message": "Stuff doesn't exist"})
+        }
+    """)
+
+    def trigger_event():
+        """
+        Make a request to the regular API that this behavior event modifies.
+        Note that the event should conform to the criteria given.
+
+        :return: a tuple of (response, body)
+        """
+
+    def validate_injected_behavior(name_and_params, response, body):
+        """
+        Validate that the named behavior was triggered as opposed to the
+        default behavior or some other behavior.
+
+        :param name_and_params: The name and parameters of the behavior to
+            trigger - this will be one of the ones specified in
+            :ivar:`name_and_params`.
+        :param response: The response from triggering the event.
+        :param body: The response body from triggering the event.
+        """
+
+    def validate_default_behavior(response, body):
+        """
+        Validate that the default behavior was triggered as opposed to
+        some other behavior.
+
+        :param response: The response from triggering the event.
+        :param body: The response body from triggering the event.
+        """
+
+    def from_test_case(test_case):
```

Comment:
Technically speaking this is not a valid way to declare an `Interface`.  An interface defines the behavior on a single object, and `from_test_case` is not a method on that same object.  There should be a separate interface, `IBehaviorAPITestHelperFactory`, which describes this one method.